test/com/google/enterprise/adaptor/prebuilt/PrebuiltTransformsTest.java

```diff
@@ -478,13 +546,69 @@ public void testMoveFromParamToParam() {
   }
 
   @Test
+  public void testMoveParamToParamOverwriteFalse() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("overwrite", "false");
+    config.put("1.from", "colour");
+    config.put("1.from.keyset", "params");
+    config.put("1.to", "color");
+    config.put("1.to.keyset", "params");
+    config = Collections.unmodifiableMap(config);
+
+    MetadataTransform transform = PrebuiltTransforms.moveMetadata(config);
+
+    final Metadata metadataGolden = new Metadata().unmodifiableView();
+    final Map<String, String> paramsGolden;
+    {
+      Map<String, String> golden = new HashMap<String, String>();
+      golden.put("color", "red");
+      paramsGolden = Collections.unmodifiableMap(golden);
+    }
+    Metadata metadata = new Metadata();
+    Map<String, String> params = new HashMap<String, String>();
+    params.put("colour", "black");
+    params.put("color", "red");
+    transform.transform(metadata, params);
+    assertEquals(metadataGolden, metadata);
+    assertEquals(paramsGolden, params);
+  }
+
+  @Test
+  public void testMoveParamToParamOverwriteTrue() {
+    Map<String, String> config = new HashMap<String, String>();
+    config.put("overwrite", "true");
+    config.put("1.from", "colour");
+    config.put("1.from.keyset", "params");
+    config.put("1.to", "color");
+    config.put("1.to.keyset", "params");
+    config = Collections.unmodifiableMap(config);
+
+    MetadataTransform transform = PrebuiltTransforms.moveMetadata(config);
+
+    final Metadata metadataGolden = new Metadata().unmodifiableView();
+    final Map<String, String> paramsGolden;
+    {
+      Map<String, String> golden = new HashMap<String, String>();
+      golden.put("color", "black");
```

Comment:
I was wrestling with this. I *could* try to avoid the delete if the move was a no-op, but overwrite == false for params is just messed up enough that I am not sure it is worth the effort.